desktop-widgets/mainwindow.h

```diff
@@ -217,6 +219,8 @@ public
 	UpdateManager *updateManager;
 	LocationInformationWidget *diveSiteEdit;
 
+	ChartWidget *statsChartWidget;
```

Comment:
I don't understand the point of this member variable. It is set but never read and it is not initialized.

If the dialog is supposed modal, you could simply use a local variable:
```
	ChartWiddget chart;
	chart.show()
```
If the dialog is not supposed to be modal, but you can have multiple dialogs, you could use a local pointer:
```
	ChartWidget *chart = new Chart(this);
	chart->show();
```
If the dialog is non-modal and there is only supposed to be one dialog at a time, then this member variable is needed, but has to be initialized to null. Or use the magic of smart pointers:
```
	std::unique_ptr<ChartWidget> chart;
```
That will automatically set to null on creation, free on destruction and with a simple
```
	chart.reset(new ChartWidget(this))
```
will destroy the old object (if any) and replace by a new one.